### DISCUSSION TITLE
getStaticProps unified Nextjs error must use import to use ES Modules

### DISCUSSION THREAD

**Discussion by crisjumbo** [unifiedjs/unified#152] [Q&A] [answered]
<p>So I am creating a static blog and using unified to compile the Markdown onto Html. All goes well the only error is when I require('unified') module.</p>
</br>
<p>The error suggested me to 1.rename the index.js (in unified package) to index.cjs and use import instead or 2.in the package.json(in unified) remove the types: module<p/>
</br>
I did both but it didn't fix my problem. Here is the blog from where I am getting the code: 
<a href="https://dev.to/sagar/building-a-blog-with-next-js-253">Blog Post</a> and here the error:
</br>
<code>
Error: Must use import to load ES Module: /path/node_modules/unified/index.js
require() of ES modules is not supported.
require() of /path/node_modules/unified/index.js from /path/server/pages/blog/[slug].js is an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which defines all .js files in that package scope as ES modules.
Instead rename index.js to end in .cjs, change the requiring code to use import(), or remove "type": "module" from /path/node_modules/unified/package.json.
</code>

**Comment by ChristianMurphy** [comment, marked as answer]
Unified 10 uses ESM https://github.com/unifiedjs/unified/releases/tag/10.0.0 this is a planned part of the latest release https://github.com/unifiedjs/unified/issues/121
For an introduction to using ESM packages see https://gist.github.com/sindresorhus/a39789f98801d908bbc7ff3ecc99d99c
In particular the workaround for Next js https://gist.github.com/sindresorhus/a39789f98801d908bbc7ff3ecc99d99c#gistcomment-3760583
Next is working on fixing ESM support, and the fixes should part of the next release https://github.com/vercel/next.js/pull/27069

**Reply by ChristianMurphy** [reply]
Update: Next 11.1 comes with better built in support for ESM https://nextjs.org/blog/next-11-1#es-modules-support
